src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.Parse.cs

```diff
@@ -126,5 +128,34 @@ public abstract partial class JsonNode
             JsonElement element = JsonElement.ParseValue(utf8Json, documentOptions);
             return JsonNodeConverter.Create(element, nodeOptions);
         }
+
+        /// <summary>
+        ///   Parse a <see cref="Stream"/> as UTF-8 encoded data representing a single JSON value into a
+        ///   <see cref="JsonNode"/>.  The Stream will be read to completion.
+        /// </summary>
+        /// <param name="utf8Json">JSON text to parse.</param>
+        /// <param name="nodeOptions">Options to control the node behavior after parsing.</param>
+        /// <param name="documentOptions">Options to control the document behavior during parsing.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>
+        ///   A <see cref="Task"/> to produce a <see cref="JsonNode"/> representation of the JSON value.
+        /// </returns>
+        /// <exception cref="JsonException">
+        ///   <paramref name="utf8Json"/> does not represent a valid single JSON value.
+        /// </exception>
+        public static async Task<JsonNode?> ParseAsync(
+            Stream utf8Json,
+            JsonNodeOptions? nodeOptions = null,
+            JsonDocumentOptions documentOptions = default,
+            CancellationToken cancellationToken = default)
+        {
+            if (utf8Json is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(utf8Json));
+            }
+
+            JsonDocument document = await JsonDocument.ParseAsync(utf8Json, documentOptions, cancellationToken).ConfigureAwait(false);
```

Comment:
The public `JsonDocument.ParseAsync` methods return `JsonDocument` instances that use rented buffers, which need to be properly disposed at the end of their lifetime. Because the document will be encapsulated by the non-disposable `JsonNode` this is not possible. 

You should instead be using `JsonDocument` instances as returned by the internal `ParseUnrented` method. Because `ParseUnrented` doesn't accept streams, you might consider adding an internal `JsonDocument.ParseAsyncCoreUnrented` method that drains the stream to a buffer and passes that to `ParseUnrented`.

Reply:
For context, here's the `ParseUnrented` method:

https://github.com/dotnet/runtime/blob/d56723bdae5ba22f907e5bff583c71a0c9a3ef8c/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs#L707-L710

and here's the `ParseAsyncCore` method that uses rented buffers:

https://github.com/dotnet/runtime/blob/d56723bdae5ba22f907e5bff583c71a0c9a3ef8c/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs#L211-L229